wdl-format/src/format/task.rs

```diff
@@ -0,0 +1,421 @@
+
+        let runtime_keyword = self
+            .syntax()
+            .children_with_tokens()
+            .find(|c| c.kind() == SyntaxKind::RuntimeKeyword)
+            .expect("Runtime section should have a runtime keyword");
+        state.indent(buffer)?;
+        buffer.push_str("runtime");
+        format_inline_comment(&runtime_keyword, buffer, state, true)?;
+
+        let open_brace = self
+            .syntax()
+            .children_with_tokens()
+            .find(|c| c.kind() == SyntaxKind::OpenBrace)
+            .expect("Runtime section should have an open brace");
+        format_preceding_comments(&open_brace, buffer, state, true)?;
+        // Open braces should ignore the "+1 rule" followed by other interrupted
+        // elements.
+        if state.interrupted() {
+            state.reset_interrupted();
+            state.indent(buffer)?;
+        } else {
+            buffer.push_str(SPACE);
+        }
+        buffer.push('{');
+        format_inline_comment(&open_brace, buffer, state, false)?;
+
+        state.increment_indent();
+
+        for item in self.items() {
+            item.format(buffer, state)?;
+        }
+
+        state.decrement_indent();
+
+        let close_brace = self
+            .syntax()
+            .children_with_tokens()
+            .find(|c| c.kind() == SyntaxKind::CloseBrace)
+            .expect("Runtime section should have a close brace");
+        format_preceding_comments(&close_brace, buffer, state, true)?;
+        state.indent(buffer)?;
+        buffer.push('}');
+        format_inline_comment(&self.syntax_element(), buffer, state, false)?;
+
+        Ok(())
+    }
+
+    fn syntax_element(&self) -> SyntaxElement {
+        SyntaxElement::Node(self.syntax().clone())
+    }
+}
+
+impl Formattable for RequirementsItem {
+    fn format(&self, buffer: &mut String, state: &mut FormatState) -> Result<()> {
+        format_preceding_comments(&self.syntax_element(), buffer, state, false)?;
+
+        let name = self.name();
+        state.indent(buffer)?;
+        name.format(buffer, state)?;
+        format_inline_comment(&name.syntax_element(), buffer, state, true)?;
+
+        let colon = self
+            .syntax()
+            .children_with_tokens()
+            .find(|c| c.kind() == SyntaxKind::Colon)
+            .expect("Requirements item should have a colon");
+        format_preceding_comments(&colon, buffer, state, true)?;
+        if state.interrupted() {
+            state.reset_interrupted();
+            state.indent(buffer)?;
+        }
+        buffer.push(':');
+        format_inline_comment(&colon, buffer, state, true)?;
+
+        let expr = self.expr();
+        format_preceding_comments(&expr.syntax_element(), buffer, state, true)?;
+        state.space_or_indent(buffer)?;
+        expr.format(buffer, state)?;
+        format_inline_comment(&self.syntax_element(), buffer, state, false)?;
+
+        Ok(())
+    }
```

Comment:
Can this logic be abstracted out? It would be the same for `RequirementsItem`/`RuntimeItem`/`HintsItem` right?

Reply:
@peterhuene how would you recommend going about this? I believe @adthrasher is correct, `RequirementsItem`. `RuntimeItem`, and `HintsItem` should receive identical formatting, but they are defined separately. Should there be a common representation for these 3 items?

Reply:
One way to do that would be to define a trait (that is a supertrait of `AstNode`) that returns what's needed from `self` here, such as `name()` and `expr()`.

With that, you'd implement the trait for `RequirementsItem`, `RuntimeItem`, and `HintsItem` to call the appropriate `name` and `expr.

Then replace the bodies here with a generic helper that takes `T` constrained by the new trait.

Reply:
Is it ok if we leave this for another PR? I'd like to get this merged in soon. @peterhuene that's good advice, I'll work on it in another branch. Related, I think I can do the same in some other places as well. Mainly thinking of `meta` and `parameter` meta sections which are dang-near identical. Same maybe with `input` and `output`? I'll have to play around with it a bit.